<a href="https://colab.research.google.com/github/SyedSameena18/CodeAlpha_Exploratory-Data-Analysis-EDA-/blob/main/CODEALPHA_TASK2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## ***EXPLORATORY DATA ANALYSIS OF GLOBAL TECH LAYOFFS***

IMPORTNG LIBRARIES -

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

LOADING DATASET -

In [ ]:
import zipfile
with zipfile.ZipFile('/content/Layoffs.zip', 'r') as zip_ref:
    zip_ref.extractall('/content')

In [ ]:
df = pd.read_csv('/content/Layoffs.zip')


DATASET OVERVIEW -

In [ ]:
df.shape

(4453, 11)

In [ ]:
df.head()

,company,location,total_laid_off,date,percentage_laid_off,industry,source,stage,funds_raised,country,date_added
0,Pleo,"Copenhagen, Non-U.S.",50.0,6/12/2026,NaN,Finance,https://tech.eu/2026/06/12/pleo-undertakes-fre...,Series C,428.0,Denmark,6/12/2026
1,ServiceNow,SF Bay Area,54.0,6/11/2026,NaN,Support,https://www.siliconvalley.com/2026/06/11/tech-...,Post-IPO,84.0,United States,6/12/2026
2,Opendoor,"Bengaluru, Non-U.S.",250.0,6/10/2026,NaN,Real Estate,https://techcrunch.com/2026/06/10/opendoors-in...,Post-IPO,1900.0,India,6/12/2026
3,Shopee,"Jakarta, Non-U.S.",NaN,6/10/2026,0.08,Food,https://www.straitstimes.com/business/seas-sho...,Unknown,NaN,Indonesia,6/10/2026
4,Salesforce,SF Bay Area,86.0,6/9/2026,NaN,Sales,https://www.businessinsider.com/salesforce-cut...,Post-IPO,65.0,United States,6/9/2026


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4453 entries, 0 to 4452
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   company              4453 non-null   object 
 1   location             4452 non-null   object 
 2   total_laid_off       2919 non-null   float64
 3   date                 4453 non-null   object 
 4   percentage_laid_off  2797 non-null   float64
 5   industry             4451 non-null   object 
 6   source               4450 non-null   object 
 7   stage                4448 non-null   object 
 8   funds_raised         3936 non-null   float64
 9   country              4451 non-null   object 
 10  date_added           4453 non-null   object 
dtypes: float64(3), object(8)
memory usage: 382.8+ KB


DATA CLEANING -

In [ ]:
df.isnull().sum()

,0
company,0
location,1
total_laid_off,1534
date,0
percentage_laid_off,1656
industry,2
source,3
stage,5
funds_raised,517
country,2


In [ ]:
df.duplicated().sum()

np.int64(0)


No duplicate records were found in the dataset.

Therefore, no rows were removed during the duplicate cleaning process.

MISSING VALUES ANALYSIS -

In [ ]:
df.isnull().sum().sort_values(ascending=False)

,0
percentage_laid_off,1656
total_laid_off,1534
funds_raised,517
stage,5
source,3
country,2
industry,2
location,1
company,0
date,0




* The dataset contains missing values in several columns. The highest number of missing values are found in:

   percentage_laid_off

   total_laid_off

   funds_raised

* Columns such as location, industry, source, stage, and country contain only a few missing values.

CONVERTING DATE COLUMNS -



In [ ]:
df['date'] = pd.to_datetime(df['date'])
df['date_added'] = pd.to_datetime(df['date_added'])
df[['date','date_added']].head()

,date,date_added
0,2026-06-12,2026-06-12
1,2026-06-11,2026-06-12
2,2026-06-10,2026-06-12
3,2026-06-10,2026-06-10
4,2026-06-09,2026-06-09


The `date` and `date_added` columns were converted from object data type to datetime format to enable time-based analysis and trend identification.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4453 entries, 0 to 4452
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   company              4453 non-null   object        
 1   location             4452 non-null   object        
 2   total_laid_off       2919 non-null   float64       
 3   date                 4453 non-null   datetime64[ns]
 4   percentage_laid_off  2797 non-null   float64       
 5   industry             4451 non-null   object        
 6   source               4450 non-null   object        
 7   stage                4448 non-null   object        
 8   funds_raised         3936 non-null   float64       
 9   country              4451 non-null   object        
 10  date_added           4453 non-null   datetime64[ns]
dtypes: datetime64[ns](2), float64(3), object(6)
memory usage: 382.8+ KB




* The dataset contains 4,453 records and 11 columns.

*   It includes information about companies, locations, industries, funding, layoff counts, and dates.

* Numerical columns:

  total_laid_off

  percentage_laid_off

  funds_raised

*  Categorical columns:

   company, location, industry, stage, country, etc.




STATISTICAL SUMMARY -

In [ ]:
df.describe(include='all')

,company,location,total_laid_off,date,percentage_laid_off,industry,source,stage,funds_raised,country,date_added
count,4453,4452,2919.000000,4453,2797.000000,4451,4450,4448,3936.000000,4451,4453
unique,2925,269,NaN,NaN,NaN,30,4194,16,NaN,66,NaN
top,Amazon,SF Bay Area,NaN,NaN,NaN,Finance,Internal memo,Post-IPO,NaN,United States,NaN
freq,19,1108,NaN,NaN,NaN,530,105,1066,NaN,2811,NaN
mean,NaN,NaN,315.957520,2023-02-24 11:43:40.166180096,0.293178,NaN,NaN,NaN,860.907240,NaN,2023-03-07 10:46:25.897148160
min,NaN,NaN,3.000000,2020-03-11 00:00:00,0.000000,NaN,NaN,NaN,0.700000,NaN,2020-03-28 00:00:00
25%,NaN,NaN,40.000000,2022-07-20 00:00:00,0.100000,NaN,NaN,NaN,55.000000,NaN,2022-07-29 00:00:00
50%,NaN,NaN,90.000000,2023-02-05 00:00:00,0.170000,NaN,NaN,NaN,175.000000,NaN,2023-02-13 00:00:00
75%,NaN,NaN,200.000000,2024-01-11 00:00:00,0.330000,NaN,NaN,NaN,483.250000,NaN,2024-01-20 00:00:00
max,NaN,NaN,30000.000000,2026-06-12 00:00:00,1.000000,NaN,NaN,NaN,121900.000000,NaN,2026-06-12 00:00:00




- The dataset contains information from 2,925 unique companies.
- The United States contributes the highest number of layoff records.
- Finance is the most frequently represented industry.
- Most layoff events were reported from the SF Bay Area.
- Post-IPO companies appear most frequently in the dataset.

In [ ]:
df.groupby('company')['total_laid_off'].sum().sort_values(ascending=False).head(10)

,total_laid_off
company,
Amazon,58124.0
Intel,43115.0
Meta,35700.0
Oracle,31294.0
Microsoft,30055.0
Dell,23650.0
Cisco,18521.0
Salesforce,16611.0
Tesla,14500.0


EXPLORATORY ANALYSIS -

In [ ]:
#Analysis 1: Top 10 Companies by Layoffs
top_companies = (
    df.groupby('company')['total_laid_off']
      .sum()
      .sort_values(ascending=False)
      .head(10)
)
top_companies

,total_laid_off
company,
Amazon,58124.0
Intel,43115.0
Meta,35700.0
Oracle,31294.0
Microsoft,30055.0
Dell,23650.0
Cisco,18521.0
Salesforce,16611.0
Tesla,14500.0


In [ ]:
#Analysis 2: Top 10 Countries by Layoffs
top_countries = (
    df.groupby('country')['total_laid_off']
      .sum()
      .sort_values(ascending=False)
      .head(10)
)
top_countries

,total_laid_off
country,
United States,657386.0
India,66289.0
Germany,32055.0
United Kingdom,23354.0
Netherlands,21575.0
Sweden,20379.0
Canada,16068.0
Israel,14619.0
Brazil,11939.0


In [ ]:
#Analysis 3: Top Industries by Layoffs
top_industries = (
    df.groupby('industry')['total_laid_off']
      .sum()
      .sort_values(ascending=False)
      .head(10)
)
top_industries

,total_laid_off
industry,
Other,119761.0
Retail,106706.0
Hardware,106261.0
Consumer,97007.0
Finance,67982.0
Transportation,66002.0
Food,52101.0
Healthcare,39462.0
Infrastructure,24835.0


In [ ]:
#Analysis 4: Layoffs Trend by Year
df['year'] = df['date'].dt.year
df.groupby('year')['total_laid_off'].sum()

,total_laid_off
year,
2020,80998.0
2021,15823.0
2022,165269.0
2023,264320.0
2024,152922.0
2025,124636.0
2026,118312.0


In [ ]:
#Analysis 5: Funding vs Layoffs
df[['total_laid_off', 'funds_raised']].corr()

,total_laid_off,funds_raised
total_laid_off,1.000000,0.119249
funds_raised,0.119249,1.000000


In [ ]:
#Analysis 6: Companies with 100% Layoffs
df[df['percentage_laid_off'] == 1][
    ['company', 'industry', 'country']
].head(20)

,company,industry,country
28,Picnic,Food,United States
48,Parker,Finance,United States
90,Pepper Pay,Finance,United States
96,NeuroPixel.AI,Retail,India
103,Covrzy,Finance,India
104,Rec Room,Consumer,United States
105,Yupp,AI,United States
108,Quandoo,Food,Germany
117,FranShares,Finance,United States
170,Entropy,Crypto,United States



1. The dataset contains 4,453 layoff records from 2,925 companies.
2. The United States accounts for the highest number of layoff events.
3. Certain industries experienced significantly higher layoffs than others.
4. Layoffs increased sharply during specific periods.
5. Several companies laid off 100% of their workforce.

# Conclusion

This exploratory data analysis examined global layoff trends across companies, industries, and countries. The analysis revealed significant workforce reductions concentrated in specific sectors and regions. The findings provide insights into how economic conditions and organizational strategies have influenced employment trends worldwide.